lib_xua/src/core/buffer/decouple/decouple.xc

```diff
@@ -170,11 +170,7 @@ static inline void _send_sample_4(chanend c_mix_out, int ch)
         mult = multOutPtr[ch];
     }
     {h, l} = macs(mult, sample, 0, 0);
-    h <<= 3;
-#if (STREAM_FORMAT_OUTPUT_RESOLUTION_32BIT_USED == 1)
-    h |= (l >>29) & 0x7; // Note: This step is not required if we assume sample depth is 24bit (rather than 32bit)
-                            // Note: We need all 32bits for Native DSD
-#endif
+    asm("lextract %0,%1,%2,%3,32":"=r"(h):"r"(h),"r"(l),"r"(3));
```

Comment:
Thanks for the update, can we leave the xc code in as comment for readability please, just so people who are not familiar with the lextract inst know what the intention is.

Reply:
(3) should be (32+3)?

Reply:
Yes, it should be 29, I only understood that on a flight to my holiday 😆

Reply:
@xross what part of the code do you want me to keep as a comment?
`h <<= 3; h |= (l >>29) & 0x7;` with some message about xs1 vs xs2?